google_maps.py

```diff
@@ -1,61 +1,47 @@
-        origin_address = prop['displayable_address'].replace(' ', '+')
-
-        for address in addresses:
-            email_message_body += address + ' travel: ' + '\n'
-            destination_address = addresses[address].replace(' ', '+')
-            url_driving = '{0}?origin={1}&destination={2}'.format(BASE_GOOGLE_MAPS_URI, origin_address, destination_address)
-            dict_data = send_request(url_driving)
-            email_message_body += ('Driving: {0} ({1})'.format(dict_data['routes'][0]['legs'][0]['duration']['text'],
-                                                               dict_data['routes'][0]['legs'][0]['distance']['text']) + '\n')
-            url_transit = '{0}?origin={1}&destination={2}&mode=transit'.format(BASE_GOOGLE_MAPS_URI,
-                                                                               origin_address, destination_address)
-            dict_data = send_request(url_transit)
-            email_message_body += 'Transit: {0} ({1})'.format(dict_data['routes'][0]['legs'][0]['duration']['text'],
-                                              dict_data['routes'][0]['legs'][0]['distance']['text']) + '\n'
-        email_message_body += '\n'
-    # send email notifying that new properties have been found
-    print(email_message_body)
-    emailer.send_notification_emails(email_message_body)
-
-
-def send_request(url):
-    """
-    Google Maps limits searches to 10 per second.
-    If limit is reached, wait 2 seconds and retry
-    :param url:
-    :return:
-    """
-    response = requests.get(url)
-    dict_data = json.loads(response.content)
-    if dict_data['status'] == 'OVER_QUERY_LIMIT':
-        time.sleep(2)
-        response = requests.get(url)
-        dict_data = json.loads(response.content)
+def collect_commute_times(properties, house_members):
+    properties_commute_times = []
+    for property in properties:
+        property_commutes = {'property_address': property['displayable_address']}
+        for house_member in house_members:
+            property_commutes[house_member['name']] = get_time_to_work(property, house_member)
+
+        properties_commute_times.append(property_commutes)
+
+    return properties_commute_times
+
+
+def get_time_to_work(property, house_member):
+    commute_times = {}
+    for travel_mode in house_member['allowed_travel_methods']:
+        travel_parameters = {
+            'origin': property['displayable_address'],
+            'destination': house_member['commute_destination'],
+            'mode': travel_mode
+        }
+
+        commute_times[travel_mode] = query_google_distance_matrix(travel_parameters)
+
+    return commute_times
 
-    if dict_data['status'] == 'OVER_QUERY_LIMIT':
+
+def query_google_distance_matrix(query_parameters):
+    response = requests.get(url=BASE_GOOGLE_MAPS_URI, params=query_parameters)
+    commute_time_response = response.json()
+    if commute_time_response['status'] == 'OVER_QUERY_LIMIT':
```

Comment:
Also, I see you removed one of the if QUERY_OVER_LIMIT checks. There were two for a reason - occasionally having just one would still fail. In the Google suggestions page, they have [3 attempts](https://developers.google.com/maps/documentation/business/articles/usage_limits).